.github/workflows/runtime-reproducibility.yml

```diff
@@ -0,0 +1,73 @@
+name: verify runtime stack reproducibility
+
+on:
+    workflow_dispatch:
+    pull_request:
+      paths:
+        - 'node-installer/**'
+        - 'packages/**'
+
+jobs:
+  os-matrix:
+    strategy:
+      matrix:
+        os: [ubuntu-22.04, ubuntu-20.04]
+        build-target: [contrast-node-installer-image]
+    # Usually we would define the matrix outputs here, but as GitHub Actions don't seem to allow per-combination outputs,
+    # we'll write the outputs without defining them here. See https://github.com/orgs/community/discussions/17245#discussioncomment-3814009.
+    runs-on: ${{ matrix.os }}
+    permissions:
+      contents: write
+    steps:
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+      - uses: ./.github/actions/setup_nix
+        with:
+          githubToken: ${{ secrets.GITHUB_TOKEN }}
+          cachixToken: ${{ secrets.CACHIX_AUTH_TOKEN }}
+      - uses: nicknovitski/nix-develop@a2060d116a50b36dfab02280af558e73ab52427d # v1.1.0
+      - name: Build
+        id: build
+        run: |
+            nix build .#${{ matrix.build-target }} --option substituters https://cache.nixos.org --builders ""
+            reference_checksum="$(cat result/index.json | jq  -r '.manifests[0].digest')"
+            echo "reference-checksum-${{ matrix.os }}-${{ matrix.build-target}}=$reference_checksum" >> "$GITHUB_OUTPUT"
+
+            nix build .#${{ matrix.build-target }} --rebuild --option substituters https://cache.nixos.org --builders "" -o rebuild
+            rebuild_checksum="$(cat rebuild/index.json | jq  -r '.manifests[0].digest')"
+            echo "rebuild-checksum-${{ matrix.os }}-${{ matrix.build-target}}=$rebuild_checksum" >> "$GITHUB_OUTPUT"
+      - name: Upload Build Artifacts
+        uses: actions/upload-artifact@65462800fd760344b1a7b4382951275a0abb4808 # v4.3.3
+        with:
+          name: ${{ matrix.build-target }}-${{ matrix.os }}
+          path: result
+      - name: Upload Build Artifacts (Rebuild)
+        uses: actions/upload-artifact@65462800fd760344b1a7b4382951275a0abb4808 # v4.3.3
+        with:
+          name: ${{ matrix.build-target }}-${{ matrix.os }}-rebuild
+          path: rebuild
+
+  collect-checksums:
+    runs-on: ubuntu-22.04
+    permissions:
+      contents: read
+    needs: os-matrix
+    steps:
+      - name: Collect checksums
+        id: collect
+        shell: python
+        run: |
+          import json, os
+          seen = {}
+          with open(os.getenv("GITHUB_OUTPUT")) as f:
+              for line in f:
+                  matrix, checksum = line.strip().split("=")
+                  if not checksum in seen:
+                      seen[checksum] = []
+                  seen[checksum].append(matrix)
+
+          if len(seen) > 1:
+            print("At least one checksum mismatched:")
+            print(json.dumps(seen, indent=2))
+            exit(1)
+
+          print("All checksums were equal")
```

Comment:
why do we need python here? check out https://github.com/edgelesssys/constellation/blob/main/.github/workflows/reproducible-builds.yml#L191 for a concise way of doing this with bash.

Reply:
We don't have files, and if we want files, we'd need to write them from the output here first, which would require us to parse it, etc., or use artifact sharing between the jobs - both incurring quite a lot of overhead if done in bash

I think this is more readable, too. But that is my personal preference